contracts/strategies/harvest/IdleHarvestStrategy.sol

```diff
@@ -0,0 +1,240 @@
+
+        ERC20Upgradeable.__ERC20_init("Idle Harvest Strategy Token", string(abi.encodePacked("idleHS", underlyingToken.symbol())));
+        //------//-------//
+
+        uniswapV2Router02 = IUniswapV2Router02(_uniswapV2Router02);
+        uniswapRouterPath = _routerPath;
+
+        transferOwnership(_owner);
+        lastIndexedTime = block.timestamp;
+    }
+
+    /// @notice redeem the rewards. Claims reward as per the _extraData
+    /// @return rewards amount of reward that is deposited to vault
+    function redeemRewards(bytes calldata) external override onlyIdleCDO returns (uint256[] memory rewards) {
+        IRewardPool(rewardPool).getReward();
+        rewards = new uint256[](1);
+        rewards[0] = IERC20Detailed(govToken).balanceOf(address(this));
+        IERC20Detailed(govToken).safeTransfer(msg.sender, rewards[0]);
+    }
+
+    /// @notice unused in harvest strategy
+    function pullStkAAVE() external pure override returns (uint256) {
+        return 0;
+    }
+
+    /// @notice return the price from the strategy token contract
+    /// @return price
+    function price() public view override returns (uint256) {
+        return IHarvestVault(strategyToken).getPricePerFullShare();
+    }
+
+    /// @notice Get the reward token
+    /// @return array of reward token
+    function getRewardTokens() external view override returns (address[] memory) {
+        address[] memory govTokens = new address[](1);
+        govTokens[0] = govToken;
+        return govTokens;
+    }
+
+    function getApr() external view returns (uint256) {
+        return lastApr;
+    }
+
+    /// @notice update last saved apr
+    /// @param _amount amount of underlying tokens to mint/redeem
+    function _updateApr(int256 _amount) internal {
+        uint256 rawBalance = IRewardPool(rewardPool).balanceOf(address(this));
+        uint256 expectedUnderlyingAmount = (price() * rawBalance) / oneToken;
+        uint256 _lastIndexAmount = lastIndexAmount;
+        if (lastIndexAmount > 0) {
+            uint256 diff = expectedUnderlyingAmount > _lastIndexAmount ? expectedUnderlyingAmount - _lastIndexAmount : 0;
+            uint256 gainPerc = ((diff) * 10**20) / _lastIndexAmount;
+            lastApr = (YEAR / (block.timestamp - lastIndexedTime)) * gainPerc;
+        }
+        lastIndexedTime = block.timestamp;
+        lastIndexAmount = uint256(int256(expectedUnderlyingAmount) + _amount);
+    }
+
+    /// @notice Redeem Tokens
+    /// @param _amount amount of strategy tokens to redeem
+    /// @return Amount of underlying tokens received
+    function redeem(uint256 _amount) external override onlyIdleCDO returns (uint256) {
+        if (_amount > 0) {
+            return _redeem(_amount, price());
+        }
+    }
+
+    /// @notice Redeem Tokens
+    /// @param _amount amount of underlying tokens to redeem
+    /// @return Amount of underlying tokens received
+    function redeemUnderlying(uint256 _amount) external returns (uint256) {
```

Comment:
this method should be protected using `onlyCDO` right?

Reply:
otherwise the receiver of funds in `_redeem` is `msg.sender`. this should not happen